server/msgpack/server.go

```diff
@@ -89,11 +96,15 @@ func (s *handler) Handle(conn net.Conn) {
 		metric := it.Metric()
 		policiesList := it.PoliciesList()
 		if err := s.aggregator.AddMetricWithPoliciesList(metric, policiesList); err != nil {
-			s.log.WithFields(
-				log.NewField("metric", metric.String()),
-				log.NewField("policies", policiesList),
-				log.NewErrField(err),
-			).Errorf("error adding metric with policies")
+			// We sample the error log here because the error rate may scale with
+			// the metrics incoming rate and consume lots of cpu cycles.
+			if s.rand.Float64() < s.errLogSamplingRate {
```

Comment:
We might want to look at `zap` at some point, we would get this for free with them.

Reply:
Yeah, I think there's an open PR in m3x that sets the zap logger on the instrument options but the last update was 20 days ago. I'd be happy to use it when it's ready but this will get me unblocked.